### ISSUE TITLE
Update example file `imdb_lstm.py` to use dropout in RNNs

### ISSUE THREAD

**Issue by yaringal** [fchollet/keras#1786]
I forgot to update the example file `imdb_lstm.py` to use dropout in RNNs as part of the pull request.
I would suggest updating it to:
> print('Build model...')
model = Sequential()
model.add(Embedding(max_features, 128, input_length=maxlen, dropout=0.5))
model.add(LSTM(128, dropout_W=0.5, dropout_U=0.5))
model.add(Dropout(0.5))
...
model.fit(X_train, y_train, batch_size=batch_size, nb_epoch=15, validation_data=(X_test, y_test), show_accuracy=True)

This is what the results look like before dropout:
> Train on 20000 samples, validate on 5000 samples
Epoch 1/15
20000/20000 [==============================] - 42s - loss: 0.4705 - acc: 0.7728 - val_loss: 0.3841 - val_acc: 0.8310
Epoch 2/15
20000/20000 [==============================] - 42s - loss: 0.2676 - acc: 0.8935 - val_loss: 0.3634 - val_acc: 0.8368
Epoch 3/15
20000/20000 [==============================] - 42s - loss: 0.1642 - acc: 0.9410 - val_loss: 0.4106 - val_acc: 0.8336
Epoch 4/15
20000/20000 [==============================] - 42s - loss: 0.0913 - acc: 0.9700 - val_loss: 0.5181 - val_acc: 0.8234
Epoch 5/15
20000/20000 [==============================] - 42s - loss: 0.0734 - acc: 0.9762 - val_loss: 0.5570 - val_acc: 0.8300
Epoch 6/15
20000/20000 [==============================] - 42s - loss: 0.0419 - acc: 0.9865 - val_loss: 0.7032 - val_acc: 0.8236
Epoch 7/15
20000/20000 [==============================] - 42s - loss: 0.0325 - acc: 0.9891 - val_loss: 0.7419 - val_acc: 0.8260
Epoch 8/15
20000/20000 [==============================] - 42s - loss: 0.0210 - acc: 0.9935 - val_loss: 0.8792 - val_acc: 0.8264
Epoch 9/15
20000/20000 [==============================] - 42s - loss: 0.0304 - acc: 0.9897 - val_loss: 0.8041 - val_acc: 0.8174
Epoch 10/15
20000/20000 [==============================] - 42s - loss: 0.0173 - acc: 0.9953 - val_loss: 0.9233 - val_acc: 0.8240
Epoch 11/15
20000/20000 [==============================] - 42s - loss: 0.0087 - acc: 0.9974 - val_loss: 0.9507 - val_acc: 0.8318
Epoch 12/15
20000/20000 [==============================] - 42s - loss: 0.0314 - acc: 0.9906 - val_loss: 0.8402 - val_acc: 0.8174
Epoch 13/15
20000/20000 [==============================] - 42s - loss: 0.0140 - acc: 0.9953 - val_loss: 0.9607 - val_acc: 0.8238
Epoch 14/15
20000/20000 [==============================] - 42s - loss: 0.0070 - acc: 0.9980 - val_loss: 1.0835 - val_acc: 0.8260
Epoch 15/15
20000/20000 [==============================] - 42s - loss: 0.0034 - acc: 0.9988 - val_loss: 1.0937 - val_acc: 0.8174
5000/5000 [==============================] - 2s     
Test score: 1.09374942386
Test accuracy: 0.8174

And this is with dropout:
> Train on 20000 samples, validate on 5000 samples
Epoch 1/15
20000/20000 [==============================] - 45s - loss: 0.6394 - acc: 0.6253 - val_loss: 0.5157 - val_acc: 0.7760
Epoch 2/15
20000/20000 [==============================] - 45s - loss: 0.5481 - acc: 0.7310 - val_loss: 0.4186 - val_acc: 0.8206
Epoch 3/15
20000/20000 [==============================] - 45s - loss: 0.4990 - acc: 0.7649 - val_loss: 0.4269 - val_acc: 0.8170
Epoch 4/15
20000/20000 [==============================] - 45s - loss: 0.4808 - acc: 0.7808 - val_loss: 0.4056 - val_acc: 0.8182
Epoch 5/15
20000/20000 [==============================] - 45s - loss: 0.4579 - acc: 0.7937 - val_loss: 0.3991 - val_acc: 0.8176
Epoch 6/15
20000/20000 [==============================] - 45s - loss: 0.4320 - acc: 0.8108 - val_loss: 0.3816 - val_acc: 0.8308
Epoch 7/15
20000/20000 [==============================] - 45s - loss: 0.4191 - acc: 0.8171 - val_loss: 0.4286 - val_acc: 0.8154
Epoch 8/15
20000/20000 [==============================] - 45s - loss: 0.3956 - acc: 0.8334 - val_loss: 0.3768 - val_acc: 0.8316
Epoch 9/15
20000/20000 [==============================] - 45s - loss: 0.3823 - acc: 0.8355 - val_loss: 0.3767 - val_acc: 0.8336
Epoch 10/15
20000/20000 [==============================] - 45s - loss: 0.3706 - acc: 0.8433 - val_loss: 0.3679 - val_acc: 0.8348
Epoch 11/15
20000/20000 [==============================] - 45s - loss: 0.3577 - acc: 0.8482 - val_loss: 0.3702 - val_acc: 0.8328
Epoch 12/15
20000/20000 [==============================] - 45s - loss: 0.3435 - acc: 0.8543 - val_loss: 0.3741 - val_acc: 0.8338
Epoch 13/15
20000/20000 [==============================] - 45s - loss: 0.3430 - acc: 0.8552 - val_loss: 0.3705 - val_acc: 0.8302
Epoch 14/15
20000/20000 [==============================] - 45s - loss: 0.3266 - acc: 0.8627 - val_loss: 0.3655 - val_acc: 0.8422
Epoch 15/15
20000/20000 [==============================] - 45s - loss: 0.3221 - acc: 0.8648 - val_loss: 0.3704 - val_acc: 0.8402
5000/5000 [==============================] - 2s     
Test score: 0.37036307745
Test accuracy: 0.8402

Should I open another pull request?

**Comment by fchollet** [comment]
Sure, sounds good!
